### PR TITLE
feature/EEMD-175_custom_radio_group

### PR DESCRIPTION
**Pull request by juantobonq**
Creation Radio group

image

https://user-images.githubusercontent.com/101196521/186755414-3df331ad-8d54-4e78-8d9f-0add7462056d.mp4

### CODE REVIEW

**Review by ks-condor — DISMISSED** [review]

File: app/build.gradle.kts [hunk 1]
```diff
@@ -50,4 +54,8 @@ dependencies {
             strictly("1.0.0-beta09")
         }
     }
+
+    androidTestImplementation("androidx.compose.ui:ui-test:1.1.0")
+    androidTestImplementation("androidx.compose.ui:ui-test-junit4:1.1.0")
+    debugImplementation("androidx.compose.ui:ui-test-manifest:1.1.0")
```
Comment (bsarias) [inline app/build.gradle.kts:60]: add dependencies strings in the buildSrc

File: app/src/androidTest/java/co/condorlabs/emerald/emeraldgroup/EmeraldGroupButtonScreenExampleTest.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+@ExperimentalTestApi
+@RunWith(AndroidJUnit4::class)
+class EmeraldGroupButtonScreenExampleTest {
+
+    @get:Rule
+    val composeRule = createAndroidComposeRule<BaseTestingActivity>()
+
+    @Test
+    fun givenItemsWhenRadioGroupIsPressedThenShouldChangeTheValueOfTheItemYES() {
```
Comment (bsarias) [inline app/src/androidTest/java/co/condorlabs/emerald/emeraldgroup/EmeraldGroupButtonScreenExampleTest.kt:29]: remember we have a criteria for UI tests, please add Page Object Pattern

File: app/src/main/java/co/condorlabs/emerald/components/radiogroup/EmeraldRadioButton.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.unit.dp
+import co.condorlabs.emerald.components.button.EmeraldButtonStyle
+import co.condorlabs.emerald.components.text.EmeraldTextStyle
+import co.condorlabs.emerald.theme.EmeraldDimens
+
+@Composable
+fun EmeraldRadioButton(
+    state: EmeraldRadioButtonState,
+    selection: EmeraldRadioButtonState,
```
Comment (bsarias) [inline app/src/main/java/co/condorlabs/emerald/components/radiogroup/EmeraldRadioButton.kt:29]: the selection is not an attribute of the Button, is an attribute of the Parent, we should not add this state to the button, i think we should pass just a 'selected' and should be a boolean

Comment (bsarias) [reply app/src/main/java/co/condorlabs/emerald/components/radiogroup/EmeraldRadioButton.kt:29]: what do you think about it?

File: app/src/main/java/co/condorlabs/emerald/theme/Theme.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package co.condorlabs.emerald.theme
+
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.darkColors
+import androidx.compose.material.lightColors
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.Color
+
+private val DarkColorPalette = darkColors(
```
Comment (bsarias) [inline app/src/main/java/co/condorlabs/emerald/theme/Theme.kt:10]: in wallet we do not use DarkPalette, please delete it

File: app/src/main/java/co/condorlabs/emerald/components/radiogroup/EmeraldRadioButton.kt [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.unit.dp
+import co.condorlabs.emerald.components.button.EmeraldButtonStyle
+import co.condorlabs.emerald.components.text.EmeraldTextStyle
+import co.condorlabs.emerald.theme.EmeraldDimens
+
+@Composable
+fun EmeraldRadioButton(
```
Comment (bsarias) [inline app/src/main/java/co/condorlabs/emerald/components/radiogroup/EmeraldRadioButton.kt:27]: this fun is exclusive of Emerald, this should not be implemented outside, please try to make it internal

`internal fun`

i am not sure if this works with funs but please try it :)

**Review by bsarias — CHANGES_REQUESTED** [review]

File: app/build.gradle.kts (hunk 1 above)
Comment (juantobonq) [reply app/build.gradle.kts:60]: Done

File: app/src/main/java/co/condorlabs/emerald/theme/Theme.kt (hunk 4 above)
Comment (juantobonq) [reply app/src/main/java/co/condorlabs/emerald/theme/Theme.kt:10]: Done

File: app/src/androidTest/java/co/condorlabs/emerald/emeraldgroup/EmeraldGroupButtonScreenExampleTest.kt (hunk 2 above)
Comment (juantobonq) [reply app/src/androidTest/java/co/condorlabs/emerald/emeraldgroup/EmeraldGroupButtonScreenExampleTest.kt:29]: Done

File: app/src/main/java/co/condorlabs/emerald/components/radiogroup/EmeraldRadioButton.kt (hunk 3 above)
Comment (juantobonq) [reply app/src/main/java/co/condorlabs/emerald/components/radiogroup/EmeraldRadioButton.kt:29]: Done

File: app/src/main/java/co/condorlabs/emerald/components/radiogroup/EmeraldRadioButton.kt (hunk 5 above)
Comment (juantobonq) [reply app/src/main/java/co/condorlabs/emerald/components/radiogroup/EmeraldRadioButton.kt:27]: Done

File: app/src/androidTest/java/co/condorlabs/emerald/uiteststubs/base/Page.kt [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+            true -> T::class.constructors.first().call()
+            false -> T::class.constructors.first().call(composeRule)
+        }
+        return page
+    }
+
+    fun wait(seconds: Int): Page {
+        Thread.sleep(seconds * 1000L)
+        return this
+    }
```
Comment (bsarias) [inline app/src/androidTest/java/co/condorlabs/emerald/uiteststubs/base/Page.kt:26]: this is not being used, please remove it

**Review by bsarias — COMMENTED** [review]
Plase add a Video or a GIF showing the funcionality

File: app/src/androidTest/java/co/condorlabs/emerald/uiteststubs/base/Page.kt (hunk 6 above)
Comment (juantobonq) [reply app/src/androidTest/java/co/condorlabs/emerald/uiteststubs/base/Page.kt:26]: Done

**Review by bsarias — APPROVED** [review]

**Review by ks-condor — APPROVED** [review]

**Pull request merged**